crates/nu-parser/src/parser.rs

```diff
@@ -6839,6 +6858,39 @@ fn wrap_expr_with_collect(working_set: &mut StateWorkingSet, expr: Expression) -
     )
 }
 
+// Report a warning if the first pipeline element of a block is `$in` followed by a pipe
+fn check_block_pipes_in(working_set: &mut StateWorkingSet, block: &Block) {
+    if let Some(elements) = block.pipelines.first().map(|pipeline| &pipeline.elements) {
+        // only warn if we're also piping into something
+        if elements.len() < 2 {
+            return;
+        }
+        let element = match elements.len() {
+            ..2 => return,
+            2.. => elements.first().expect("at least two elements"),
+        };
```

Comment:
Isn't the early return here redundant.
```suggestion
        let element = match elements.len() {
            ..2 => return,
            2.. => elements.first().expect("at least two elements"),
        };
```

or 
```suggestion
        if elements.len() < 2 {
            return;
        }
        let element = elements.first().expect("at least two elements");
```